data/prepro.py

```diff
@@ -111,31 +93,55 @@ def create_data_mats(data_toks, ques_inds, ans_inds, params, dtype):
         image_index[i] = i
         caption_len[i] = len(data_toks[image_id]['caption_inds'][0:max_cap_len])
         captions[i][0:caption_len[i]] = data_toks[image_id]['caption_inds'][0:max_cap_len]
-        for j in range(10):
 
+    num_rounds = 10
+    max_ques_len = params.max_ques_len
+    max_ans_len = params.max_ans_len
+
+    questions = np.zeros([num_threads, num_rounds, max_ques_len])
+    answers = np.zeros([num_threads, num_rounds, max_ans_len])
+    question_len = np.zeros([num_threads, num_rounds], dtype=np.int)
+    answer_len = np.zeros([num_threads, num_rounds], dtype=np.int)
+
+    # create questions and answers data mats
+    for i in range(num_threads):
+        image_id = list(data_toks.keys())[i]
+        for j in range(num_rounds):
             if data_toks[image_id]['dialog'][j]['question'] != -1:
                 question_len[i][j] = len(ques_inds[data_toks[image_id]['dialog'][j]['question']][0:max_ques_len])
                 questions[i][j][0:question_len[i][j]] = ques_inds[data_toks[image_id]['dialog'][j]['question']][0:max_ques_len]
             if data_toks[image_id]['dialog'][j]['answer'] != -1:
                 answer_len[i][j] = len(ans_inds[data_toks[image_id]['dialog'][j]['answer']][0:max_ans_len])
                 answers[i][j][0:answer_len[i][j]] = ans_inds[data_toks[image_id]['dialog'][j]['answer']][0:max_ans_len]
-            if dtype != 'test':
+
+    # create ground truth answer and options data mats
+    answer_index = np.zeros([num_threads, num_rounds])
+    num_rounds_list = np.full(num_threads, 10)
+    if dtype == 'test':
+        # dtype=test has options only for the last rounsd
+        options = np.zeros([num_threads, 1, 100])
+    else:
+        options = np.zeros([num_threads, num_rounds, 100])
+
+    for i in range(num_threads):
+        image_id = list(data_toks.keys())[i]
+        for j in range(num_rounds):
+            if dtype == 'test':
+                num_rounds_list[i] = data_toks[image_id]['num_rounds']
+                if j == num_rounds_list[i] - 1:
+                    options[i] = np.array(data_toks[image_id]['dialog'][j]['answer_options']) + 1
```

Comment:
Would be good to add a NOTE here saying that the list is 1-indexed for lua-torch specifically.